server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOBuilder.java

```diff
@@ -19,18 +19,43 @@
 
 package org.apache.james.blob.objectstorage;
 
+import java.time.Duration;
 import java.util.Optional;
 import java.util.function.Supplier;
 
 import org.apache.james.blob.api.BlobId;
 import org.apache.james.blob.api.BucketName;
 import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.domain.Blob;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
 public class ObjectStorageBlobsDAOBuilder {
 
+    static class DefaultPutBlobFunction implements PutBlobFunction {
+
+        private static final int MAX_RETRIES = 3;
+        private static final Duration FIRST_BACK_OFF = Duration.ofMillis(100);
+
+        private final BlobStore blobStore;
+
+        DefaultPutBlobFunction(BlobStore blobStore) {
+            this.blobStore = blobStore;
+        }
+
+        @Override
+        public void putBlob(BucketName bucketName, Blob blob) {
+            Mono.fromRunnable(() -> blobStore.putBlob(bucketName.asString(), blob))
+                .publishOn(Schedulers.elastic())
+                .retryBackoff(MAX_RETRIES, FIRST_BACK_OFF)
```

Comment:
are you retrying regardless of the error type?

Reply:
It can fail for a good reason (and then retrying would be a waste of resources).

Reply:
I made them just retry if the bucket doesn't exist. is that okay to you?

Reply:
what if putBlob throws an exception that is not related?